fastserde/avro-fastserde/src/main/java/com/linkedin/avro/fastserde/FastDeserializerGeneratorBase.java

```diff
@@ -97,7 +97,7 @@ protected ListIterator<Symbol> actionIterator(FieldAction action) {
     while (actionIterator.hasNext()) {
       Symbol symbol = actionIterator.next();
 
-      if(Symbol.Kind.REPEATER.equals(symbol.kind) && "array-end".equals(((Symbol.Repeater)symbol).end.toString())) {
+      if(Symbol.Kind.REPEATER.equals(symbol.kind)) {
```

Comment:
This change is not really needed, but I feel it should include `map-end` too. Let me know if you'd like me to remove this